lib/spack/spack/directives.py

```diff
@@ -306,29 +339,32 @@ def resource(pkg, **kwargs):
     * 'placement' : (optional) gives the possibility to fine tune how the
       resource is moved into the main package stage area.
     """
-    when = kwargs.get('when', pkg.name)
-    destination = kwargs.get('destination', "")
-    placement = kwargs.get('placement', None)
-    # Check if the path is relative
-    if os.path.isabs(destination):
-        message = "The destination keyword of a resource directive can't be"
-        " an absolute path.\n"
-        message += "\tdestination : '{dest}\n'".format(dest=destination)
-        raise RuntimeError(message)
-    # Check if the path falls within the main package stage area
-    test_path = 'stage_folder_root'
-    normalized_destination = os.path.normpath(join_path(test_path, destination)
-                                              )  # Normalized absolute path
-    if test_path not in normalized_destination:
-        message = "The destination folder of a resource must fall within the"
-        " main package stage directory.\n"
-        message += "\tdestination : '{dest}'\n".format(dest=destination)
-        raise RuntimeError(message)
-    when_spec = parse_anonymous_spec(when, pkg.name)
-    resources = pkg.resources.setdefault(when_spec, [])
-    name = kwargs.get('name')
-    fetcher = from_kwargs(**kwargs)
-    resources.append(Resource(name, fetcher, destination, placement))
+    def _execute(pkg):
+        when = kwargs.get('when', pkg.name)
+        destination = kwargs.get('destination', "")
+        placement = kwargs.get('placement', None)
+        # Check if the path is relative
```

Comment:
@alalazo: is there some chance that I can convince you to use blank lines before sub-parts of long sections of code?